src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -1474,8 +1503,7 @@ public boolean createPickupRequest(final int scaledPriority)
         {
             for (final IToken<?> req : reqs)
             {
-                final IRequest<?> request = colony.getRequestManager().getRequestForToken(req);
-                if (request != null && request.getState() == RequestState.IN_PROGRESS)
+                if (colony.getRequestManager().getRequestForToken(req).getState() == RequestState.IN_PROGRESS)
```

Comment:
Is this a good idea? I had the null check here on purpose.

Reply:
I am pretty sure that null check is needed as well....
It threw exceptions previously but that was removed.

Reply:
This is a bad merge/backout of the old code. Fixing.